src/dlgMapper.cpp

```diff
@@ -230,16 +187,60 @@ void dlgMapper::slot_togglePanel()
-    glWidget->setVisible(!glWidget->isVisible());
-    mp2dMap->setVisible(!mp2dMap->isVisible());
-    if (glWidget->isVisible()) {
-        d3buttons->setVisible(true);
-    } else {
-        d3buttons->setVisible(false);
+
+    if (mpHost->mpMap->mpM && mpHost->mpMap->mpMapper) {
+        mpHost->mpMap->mpM->update();
+    }
+    if (!mpHost->mpMap->mpM) {
+        glWidget = new GLWidget(widget);
+        glWidget->setObjectName(QString::fromUtf8("glWidget"));
+
+        QSizePolicy sizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+        sizePolicy.setHorizontalStretch(0);
+        sizePolicy.setVerticalStretch(0);
+        sizePolicy.setHeightForWidth(glWidget->sizePolicy().hasHeightForWidth());
+        glWidget->setSizePolicy(sizePolicy);
+        verticalLayout_2->insertWidget(0, glWidget);
+
+        glWidget->mpMap = mpMap;
+        mpMap->mpM = mpMap->mpMapper->glWidget;
+        connect(ortho, &QAbstractButton::pressed, glWidget, &GLWidget::fullView);
+        connect(singleLevel, &QAbstractButton::pressed, glWidget, &GLWidget::singleView);
+        connect(increaseTop, &QAbstractButton::pressed, glWidget, &GLWidget::increaseTop);
+        connect(increaseBottom, &QAbstractButton::pressed, glWidget, &GLWidget::increaseBottom);
+        connect(reduceTop, &QAbstractButton::pressed, glWidget, &GLWidget::reduceTop);
+        connect(reduceBottom, &QAbstractButton::pressed, glWidget, &GLWidget::reduceBottom);
+        connect(shiftZup, &QAbstractButton::pressed, glWidget, &GLWidget::shiftZup);
+        connect(shiftZdown, &QAbstractButton::pressed, glWidget, &GLWidget::shiftZdown);
+        connect(shiftLeft, &QAbstractButton::pressed, glWidget, &GLWidget::shiftLeft);
+        connect(shiftRight, &QAbstractButton::pressed, glWidget, &GLWidget::shiftRight);
+        connect(shiftUp, &QAbstractButton::pressed, glWidget, &GLWidget::shiftUp);
+        connect(shiftDown, &QAbstractButton::pressed, glWidget, &GLWidget::shiftDown);
+        connect(showInfo, &QAbstractButton::clicked, glWidget, &GLWidget::showInfo);
+        connect(defaultView, &QAbstractButton::pressed, glWidget, &GLWidget::defaultView);
+        connect(sideView, &QAbstractButton::pressed, glWidget, &GLWidget::sideView);
+        connect(topView, &QAbstractButton::pressed, glWidget, &GLWidget::topView);
+        connect(scale, &QAbstractSlider::valueChanged, glWidget, &GLWidget::setScale);
+        connect(xRot, &QAbstractSlider::valueChanged, glWidget, &GLWidget::setXRotation);
+        connect(yRot, &QAbstractSlider::valueChanged, glWidget, &GLWidget::setYRotation);
+        connect(zRot, &QAbstractSlider::valueChanged, glWidget, &GLWidget::setZRotation);
+    }
+
+    if (mpHost->mpMap->mpM) {
+        mp2dMap->setVisible(!mp2dMap->isVisible());
+        glWidget->setVisible(!glWidget->isVisible());
+        if (glWidget->isVisible()) {
+            d3buttons->setVisible(true);
+        } else {
+            // Use timer do not let the buttons dissappear
```

Comment:
I don't understand this, what does it mean?

Reply:
I'm not sure if everyone had this issue. But for me in 4.6.1 if i change between 2d/3d Mapper view. The mapper buttons reloaded strangely sometimes (and also disappeared).
this fixes it for me.

Reply:
Does a timer of `0` also work?

Try a comment of `// workaround for buttons reloading oddly`

Reply:
No. A timer of 0 doesn't work.
If this timer could potentially create other issues I'll revert it because the buttons reloading strangely/disappearing is a minor issue (they reappear after clicking on the mapper or reloading the window)

Reply:
It should be OK.